linera-execution/src/system.rs

```diff
@@ -754,6 +752,35 @@ where
         })
     }
 
+    /// Debits an [`Amount`] of tokens from an account's balance.
+    async fn debit(
+        &mut self,
+        account: Option<&Owner>,
+        amount: Amount,
+    ) -> Result<(), SystemExecutionError> {
+        let balance = if let Some(owner) = account {
+            self.balances.get_mut(owner).await?.ok_or_else(|| {
+                SystemExecutionError::InsufficientFunding {
+                    balance: Amount::ZERO,
+                }
+            })?
+        } else {
+            self.balance.get_mut()
+        };
+
+        balance
+            .try_sub_assign(amount)
+            .map_err(|_| SystemExecutionError::InsufficientFunding { balance: *balance })?;
+
+        if let Some(owner) = account {
```

Comment:
You already check whether `account` is `Some(_)` so maybe you can extract that at the beginning and reuse?

Reply:
Ok, you're right. I missed the fact that we fallback to the chain's owner balance .